File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
#첫째줄에 n과 m을 입력받음
N, M = map(int, input().split())
#n_list의 n만큼의 배열을 만들기위한 for문(n이 5면 1 2 3 4 5이런식)
n_list = [i for i in range(1,N+1)]
#M범위까지 순회하면서
for _ in range(M):
    #i, j를 입력받음
    i, j = map(int,input().split())
    #n_list의 i-1번째부터 j-1번째까지 의 원소들을 temp에 저장
    temp = n_list[i-1:j]
    #temp리스트의 원소값들을 역순으로 만들어내는 함수
    temp.reverse()
    #n_list의 i-1번째부터 j-1번째 원소값들에 변화된 temp값을 저장
    n_list[i-1:j] = temp
    #N번째까지 순회하면서 n_list[i]번째 값들을 출력
for i in range(N):
    print(n_list[i], end = ' ')

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.io.*;
import java.lang.reflect.Array;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] str = br.readLine().split(" ");
        int N = Integer.parseInt(str[0]);
        int M = Integer.parseInt(str[1]);
        int[] arr = new int[N];
        for(int y =1; y <= N; y++) {
            arr[y - 1] = y;
        }
        for(int x = 0; x< M; x++){
            String[] str2 = br.readLine().split(" ");
            int i = Integer.parseInt(str2[0])-1;
            int j = Integer.parseInt(str2[1])-1;
            while(i < j){
                /*temp에 arr[i]번째 12345에서 1 4이면
                0번째 원소값을 temp에 저장하고, 1증가된 1번째 인덱스값에 arr[j]값을 i에 저장
                1 4이면 -> 0 3
                1. temp = 1  -> 42345 -> 42315 i 1   j 2
                2. temp = 2  -> 43315 -> 43215 i 2   j 1
                * */
                int temp = arr[i];
                arr[i++] = arr[j];
                arr[j--] = temp;
            }
        }
        for(int i = 0; i< arr.length;i++){
            System.out.print(arr[i]+ " ");
        }
    }
    }